src-js/instantsearch.js

```diff
@@ -214,8 +214,10 @@ export function init () {
   });
 
   window.addEventListener('load', renderResults);
-  searchResetButton.addEventListener('click', function () {
+  searchResetButton.addEventListener('click', function (e) {
```

Comment:
I couldn't find where the visibility for the reset button was getting set, but the behaviour of it is a bit weird. 

The reset button not only clears the input, but also hides the results. On desktop the input is still visible, but on mobile the whole search experience is closed so if you intended to just clear the input you'd have to reopen search. On the other hand there is no way to close the search on mobile without putting something in the input (which causes the reset X to appear). There is no esc key on mobile. Perhaps there should be 2 buttons, one to clear the input (I'd recommend some small grey text that said clear on the right) and one to close the search, and hide the results. I'll leave a comment on the Jira ticket to see what the design team thinks.

Reply:
I know that the instantsearch JS does some of this on its own in a way we can't influence, so our options may be somewhat limited. We can look into it though.